### ISSUE TITLE
Proposed UI Update

### ISSUE THREAD

**Issue by theochawla** [jupytercalpoly/jupyterlab-notifications#69]
I’ve included a new block icon (the notif bell with a stroke through it). I also think the notifications could be condensed by reducing the line spacing and removing the third line of text. I’ve included a text hierarchy with type size and spacing. They are in px right now, would having them in ems or something similar be more helpful? I also think the drop shadow could be a little softer and more centered under the notification to better show its upper boundary.

https://user-images.githubusercontent.com/83424843/152618353-135ed33c-0167-4fd4-9aee-423670b2c0e7.mov

<img width="793" alt="Screen Shot 2022-02-04 at 3 55 42 PM" src="https://user-images.githubusercontent.com/83424843/152618488-06cd7dae-db96-424e-854a-ea2939760459.png">

**Comment by andrii-i** [comment]
Thanks for the images, they are very helpful. 
Having all font sizes in relative units like ems would be super helpful and would make the fonts look better on screens/windows of non-standard size.

**Comment by theochawla** [comment]
Here are the type sizes in ems!
<img width="781" alt="Screen Shot 2022-02-07 at 3 26 24 PM" src="https://user-images.githubusercontent.com/83424843/152889361-a5d12c27-523b-47a9-82ca-54255200fe22.png">

**Comment by andrii-i** [comment]
Angela, let's schedule a UI implementation review for when possible. Key question: Material UI that is used in Notifications UI uses "rem", not "ems" and that's what I used in the implementation. Does it works or recommended sizes should be adjusted? 

Current look:
![image](https://user-images.githubusercontent.com/26686070/154369562-1dae3de9-4708-4be7-8442-e24dcf1b544b.png)

**Comment by andrii-i** [comment]
Wasn't able to update the design of the notification popup (bottom right corner) because we use a 3rd-party extension for it. Created issue #71 to solve down the road.

**Comment by theochawla** [comment]
These look great! The line spacing looks a little looser than in the mockup: maybe we could tighten them up? I wonder if I gave you the wrong em/rem values?

**Comment by andrii-i** [comment]
Thank you! 
Here "rem" is used while you gave "ems" so the discrepancy is probably due to that.

**Comment by theochawla** [comment]
Ok, I will get you updated rem values! Right now, do you know what the root size is (the value the rems are being based on)?

**Comment by theochawla** [comment]
If it is not specified, I think the default is 16px

**Comment by andrii-i** [comment]
I also haven't implemented small icons in notifications (on the left of the title) because to do it I'll have to change/update notification API providing some mechanism for extensions to send their logo with every message or somehow register/cache it on the server.
It's a good-looking functionality tho so I'll think about how to implement it, would create a new issue.

**Comment by theochawla** [comment]
Ok, no worries! I want to create a few different kinds of icons for different notifications, so no rush on implementing that. I will update the issue with the icons when I finish them.

**Comment by andrii-i** [comment]
> If it is not specified, I think the default is 16px

Yes, as far as I understand it's always a 16px base and then each user/browser modifies this base size based on its needs (https://mui.com/customization/typography/#font-size)

**Comment by theochawla** [comment]
That makes sense! Thank you :)

**Comment by andrii-i** [comment]
> Ok, no worries! I want to create a few different kinds of icons for different notifications, so no rush on implementing that. I will update the issue with the icons when I finish them.

It's a very interesting idea that can make a life for extension devs much easier. The problem here is that API would accommodate all kinds of extensions (and therefore message types) so it might be hard to pre-plan for icons. If we could do it though, it would be nice to have a small default library that extension devs could use. 

Were you thinking about like creating icons for broad categories?

**Comment by theochawla** [comment]
> > Ok, no worries! I want to create a few different kinds of icons for different notifications, so no rush on implementing that. I will update the issue with the icons when I finish them.
> 
> It's a very interesting idea that can make a life for extension devs much easier. The problem here is that API would accommodate all kinds of extensions (and therefore message types) so it might be hard to pre-plan for icons. If we could do it though, it would be nice to have a small default library that extension devs could use.
> 
> Were you thinking about like creating icons for broad categories?

Yes! I was thinking maybe one icon for cell execution, one for new user login, and one for any custom notification. I like the idea of a default library, though. Maybe splitting up potential notifs into categories (system updates, comments, etc.) and having at least one catch-all icon?

**Comment by theochawla** [comment]
Also: here are the type sizes in rems. Sorry for the confusion, that was totally my bad.
<img width="805" alt="Screen Shot 2022-02-16 at 3 36 58 PM" src="https://user-images.githubusercontent.com/83424843/154376100-db23f57f-90de-4234-8208-d0b61a56422a.png">

**Comment by andrii-i** [comment]
> > > Ok, no worries! I want to create a few different kinds of icons for different notifications, so no rush on implementing that. I will update the issue with the icons when I finish them.
> > 
> > 
> > It's a very interesting idea that can make a life for extension devs much easier. The problem here is that API would accommodate all kinds of extensions (and therefore message types) so it might be hard to pre-plan for icons. If we could do it though, it would be nice to have a small default library that extension devs could use.
> > Were you thinking about like creating icons for broad categories?
> 
> Yes! I was thinking maybe one icon for cell execution, one for new user login, and one for any custom notification. I like the idea of a default library, though. Maybe splitting up potential notifs into categories (system updates, comments, etc.) and having at least one catch-all icon?

Your idea sounds good and is much more manageable in terms of scope. Let's go with it. In the meantime, we can think if these broad categories are even identifiable.

**Comment by theochawla** [comment]
> > > > Ok, no worries! I want to create a few different kinds of icons for different notifications, so no rush on implementing that. I will update the issue with the icons when I finish them.
> > > 
> > > 
> > > It's a very interesting idea that can make a life for extension devs much easier. The problem here is that API would accommodate all kinds of extensions (and therefore message types) so it might be hard to pre-plan for icons. If we could do it though, it would be nice to have a small default library that extension devs could use.
> > > Were you thinking about like creating icons for broad categories?
> > 
> > 
> > Yes! I was thinking maybe one icon for cell execution, one for new user login, and one for any custom notification. I like the idea of a default library, though. Maybe splitting up potential notifs into categories (system updates, comments, etc.) and having at least one catch-all icon?
> 
> Your idea sounds good and is much more manageable in terms of scope. Let's go with it. In the meantime, we can think if these broad categories are even identifiable.

Definitely. I'll start with the new user, Jupyter & extension updates, comments, and catch-all. If we think of others, I can add them.

**Comment by andrii-i** [comment]
Looks like this with provided rem values:
![image](https://user-images.githubusercontent.com/26686070/154377529-5fa7a94a-9d7e-4ead-a18d-05dc2e3e3f80.png)

**Comment by theochawla** [comment]
It's looking better! Ok. If you are up for making these changes now, I think the timestamp could be the same size as the body type. Can we increase the "more notifications" text size to .575 rems? I also think the headings in the interface could be scaled proportionally could you make their size, maybe 1 rem. The line spacing is still looser than expected. Could you try decreasing them each by .25 rems? If you have other things to finish now, feel free to push these to next week.

**Comment by andrii-i** [comment]
> It's looking better! Ok. If you are up for making these changes now, I think the timestamp could be the same size as the body type. Can we increase the "more notifications" text size to .575 rems? I also think the headings in the interface could be scaled proportionally could you make their size, maybe 1 rem. The line spacing is still looser than expected. Could you try decreasing them each by .25 rems? If you have other things to finish now, feel free to push these to next week.

Ok, I'll get back to tweaking it next week. If it's not too much to ask, updated hierarchy would really help as I'm afraid to get lost in adjustments.

**Comment by theochawla** [comment]
> > It's looking better! Ok. If you are up for making these changes now, I think the timestamp could be the same size as the body type. Can we increase the "more notifications" text size to .575 rems? I also think the headings in the interface could be scaled proportionally could you make their size, maybe 1 rem. The line spacing is still looser than expected. Could you try decreasing them each by .25 rems? If you have other things to finish now, feel free to push these to next week.
> 
> Ok, I'll get back to tweaking it next week. If it's not too much to ask, updated hierarchy would really help as I'm afraid to get lost in adjustments.

Definitely! No problem at all. I'll get that to you today :)

**Comment by theochawla** [comment]
> > > It's looking better! Ok. If you are up for making these changes now, I think the timestamp could be the same size as the body type. Can we increase the "more notifications" text size to .575 rems? I also think the headings in the interface could be scaled proportionally could you make their size, maybe 1 rem. The line spacing is still looser than expected. Could you try decreasing them each by .25 rems? If you have other things to finish now, feel free to push these to next week.
> > 
> > 
> > Ok, I'll get back to tweaking it next week. If it's not too much to ask, updated hierarchy would really help as I'm afraid to get lost in adjustments.
> 
> Definitely! No problem at all. I'll get that to you today :)
<img width="912" alt="External Heading. Size 1 rem. Line spacing 1.25 rem. Bold
Heading. Size .75 rems, Line spacing .75 rem. Bold
Action Type. Size .625 rem, Line Spacing .75 rem. Bold
Notif text. .625 rem, Line Spacing .5 rems. Medium
X more notifications. Size .575 rems, Line Spacing .56 ems. Medium
Note: current sizing in this style guide is 1 rem = 16px
Padding around type is .625 rem" src="https://user-images.githubusercontent.com/83424843/154381405-e1348ccd-e416-4cd2-9922-1e520d0e94c0.png">

**Comment by andrii-i** [comment]
Thank you!

**Comment by andrii-i** [comment]
Latest updated from @angelachawla: 

<img width="476" alt="Screen Shot 2022-04-14 at 3 28 50 PM" src="https://user-images.githubusercontent.com/26686070/163690094-afbd3131-6a18-4820-955f-ae7631adc4e6.png">

**Comment by andrii-i** [comment]
Implemented update on font sizes:
<img width="1506" alt="Screen Shot 2022-04-19 at 11 13 00 PM" src="https://user-images.githubusercontent.com/26686070/164162765-46518ffa-3d66-41a0-9694-8fca9c816fc4.png">
